osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/misc/GetGsmDiagnosticCommandExecutorTest.java

```diff
@@ -0,0 +1,263 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.Collections;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.openmuc.jdlms.AccessResultCode;
+import org.openmuc.jdlms.AttributeAddress;
+import org.openmuc.jdlms.GetResult;
+import org.openmuc.jdlms.datatypes.DataObject;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.dlmsobjectconfig.DlmsObjectConfigService;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.dlmsobjectconfig.DlmsObjectType;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.dlmsobjectconfig.model.CommunicationMethod;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.dlmsobjectconfig.model.DlmsObject;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.utils.DlmsHelper;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.DlmsDevice;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.Protocol;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.factories.DlmsConnectionManager;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.ProtocolAdapterException;
+import org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging.DlmsMessageListener;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.BitErrorRateDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.CircuitSwitchedStatusDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.GetGsmDiagnosticRequestDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.GetGsmDiagnosticResponseDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ModemRegistrationStatusDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.PacketSwitchedStatusDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.SignalQualityDto;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class GetGsmDiagnosticCommandExecutorTest {
+
+  @InjectMocks private GetGsmDiagnosticCommandExecutor executor;
+
+  @Mock private DlmsMessageListener dlmsMessageListener;
+
+  @Mock private DlmsHelper dlmsHelper;
+
+  @Mock private DlmsObjectConfigService dlmsObjectConfigService;
+
+  @Mock private DlmsConnectionManager connectionManager;
+
+  private final DlmsDevice device = this.createDevice(Protocol.SMR_5_1, CommunicationMethod.CDMA);
+  private final long from = 1111111L;
+  private final long to = 2222222L;
+  private final int classId = 99;
+  private final String obiscode = "1.2.3.4.5.6";
```

Comment:
`obisCode` would look better as camel-cased version of the two words.

Reply:
Agreed